app/models/article.rb

```diff
@@ -64,7 +64,7 @@ class Article < ApplicationRecord
   before_save       :set_caches
   before_save       :fetch_video_duration
   before_save       :clean_data
-  after_save        :async_score_calc, if: :published
+  after_commit      :async_score_calc, if: :published, unless: :destroyed?
```

Comment:
Can you really combine booleans like this? I think you might want to use a Proc here or possibly add a guard clause to the `async_score_calc` method below.

Reply:
Yep - it works 😀

https://guides.rubyonrails.org/v5.2/active_record_callbacks.html#multiple-conditions-for-callbacks

I'm happy to change it if we feel it's better with a Proc! What do you think?

Reply:
TIL! IMO I think a single boolean would read better but it's not a huge deal for me.

Reply:
I'm already not a big fan of conditional callbacks, so I'm with @mstruve here on not combining `if:` and `unless:` for a single callback.

Reply:
Updated to a guard clause.